File: 20260604/exos/02_dict.py
# ============================================================
# Resident Evil — inventaire de Leon
# ============================================================
# Leon ramasse des munitions. L'inventaire est un dictionnaire :
# la clé est le nom de l'objet, la valeur est la quantité.
# Ajoute 4 cartouches à l'entrée "munitions".
#
# Lance : python3 02_dict.py
# ============================================================

inventaire = {"munitions": 6, "herb": 2}

# inventaire["munitions"] vaut 6 au départ.
# Ajoute 4 à cette quantité (une seule ligne).
#
#   6 + 4 = 10
#
# Résultat attendu : inventaire["munitions"] == 10
#
# Indice : inventaire["munitions"] = inventaire["munitions"] + 4

# À toi :
inventaire["munitions"] += 4

# --- Vérification (ne pas modifier) ---
assert inventaire["munitions"] == 10, f"Obtenu : {inventaire['munitions']}, attendu : 10"
print("✅ Correct !")

File: 20260723/exos/02_dict.lua
-- ============================================================
-- Resident Evil — fiche inventaire
-- ============================================================
-- Lis la quantité d'herbes, puis change le statut du sac.
--
-- Lance : lua 02_dict.lua
-- ============================================================

sac = {
  herbes = 3,
  statut = "leger",
}

herbes_lues = 0

-- 1) Lis sac["herbes"] (ou sac.herbes) → herbes_lues
-- 2) Écris sac["statut"] = "lourd"
--
-- Résultat attendu :
--   herbes_lues == 3
--   sac["statut"] == "lourd"
--
-- Indice : t.cle pour lire, t.cle = valeur pour écrire

-- À toi :
herbes_lues = sac.herbes
sac.statut = "lourd"

-- --- Vérification (ne pas modifier) ---
assert(herbes_lues == 3, "herbes_lues doit valoir 3")
assert(sac["statut"] == "lourd", "statut doit être 'lourd'")
print("✅ Correct !")
